pkg/container-collection/options.go

```diff
@@ -585,6 +581,30 @@ func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
 	}
 }
 
+// WithOCIConfigEnrichment enriches container using provided OCI config
+func WithOCIConfigEnrichment() ContainerCollectionOption {
+	return func(cc *ContainerCollection) error {
+		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
+			if container.OciConfig != nil {
+				if cm, ok := container.OciConfig.Annotations[crioContainerManagerAnnotation]; ok {
+					container.Runtime = cm
+				}
+				if cn, ok := container.OciConfig.Annotations[crioContainerNameAnnotation]; ok {
+					container.Name = cn
+				}
+				if pn, ok := container.OciConfig.Annotations[crioPodNameAnnotation]; ok {
+					container.Podname = pn
+				}
+				if ns, ok := container.OciConfig.Annotations[crioPodNamespaceAnnotation]; ok {
+					container.Namespace = ns
+				}
+			}
```

Comment:
It should also enrich with the Pod UID.

Reply:
Yes it is available but just for crio via annotation: `io.kubernetes.pod.uid`. I will go ahead and add it.